src/hooks/UseForm/useForm/useForm.ts

```diff
@@ -0,0 +1,458 @@
+    ): RegisterResult<TFieldElement> => {
+      options ??= {}
+      const fields = fieldsGraph.current.getVertex(options.coordinates) ?? {}
+      if (fieldName in fields) {
+        updateFieldsVertex(fieldName, field => ({ ...field!, options }), options.coordinates)
+      } else {
+        setFieldsVertex(
+          fieldName,
+          // @ts-expect-error options are valid because the `validate` type depends upon `isRequired`
+          {
+            name: fieldName,
+            ref: React.createRef<TFieldElement | null>(),
+            options,
+            value: undefined,
+            error: null,
+            hasBeenTouched: false,
+            hasBeenChanged: false,
+          },
+          options?.coordinates,
+        )
+      }
+
+      return {
+        ref: (element: TFieldElement) => {
+          const fields = fieldsGraph.current.getVertex(options.coordinates)
+          if (fields?.[fieldName]) {
+            // update internal value from ref if not yet set
+            if (fields[fieldName]!.value === undefined) {
+              const defaultValue = getElementDefaultValue<TData>(element)
+              updateFieldsVertex(
+                fieldName,
+                field => ({
+                  ...field!,
+                  value: defaultValue,
+                  ref: { ...field?.ref, current: element },
+                }),
+                options.coordinates,
+              )
+            } else {
+              updateFieldsVertex(
+                fieldName,
+                field => ({ ...field!, ref: { ...field?.ref, current: element } }),
+                options.coordinates,
+              )
+            }
+          }
+        },
+        onChange: event => {
+          const fields = fieldsGraph.current.getVertex(options.coordinates)!
+
+          if (fieldName in fields) {
+            if (!fields[fieldName]!.hasBeenChanged) {
+              change(fieldName, options.coordinates)
+            }
+
+            const newValue = getOnChangeValue<TData>(event)
+            updateFieldsVertex(
+              fieldName,
+              field => ({ ...field!, value: newValue }),
+              options?.coordinates,
+            )
+          }
+        },
+        onFocus: () => {
+          const fields = fieldsGraph.current.getVertex(options.coordinates)!
+
+          if (fieldName in fields) {
+            if (!fields[fieldName]!.hasBeenTouched) {
+              touch(fieldName, options?.coordinates)
+            }
+          }
+        },
+        onBlur: () => {
+          updateErrorsVertexFieldError(fieldName, options?.coordinates)
+        },
+      }
+    },
+    [change, setFieldsVertex, touch, updateErrorsVertexFieldError, updateFieldsVertex],
+  )
+
+  return {
+    register,
+    errors,
+    touched,
+    changed,
+    hasBegun,
+    hasChangedWithoutSubmit,
+    handleSubmit,
+  }
+}
+
+export * from './types/public'
+export default useForm
```

Comment:
These are already re-exported from `index.ts`

```suggestion
}

export default useForm
```